src/lib/crypto/kimchi_backend/gadgets/foreign_field.ml

```diff
@@ -204,6 +278,17 @@ end = struct
       in
       of_limbs (limb_vars.(0), limb_vars.(1), limb_vars.(2))
 
+    let const_of_bignum_bigint (type field)
```

Comment:
do these automatically generate generic gates to check the constant values?

Reply:
They will get generated by snarky when the consts are used.

If you want to automatically generate them at a certain point in the circuit (e.g. at the start) then you'd use
`checked_const_of_bignum_bigint`